src/includes/class-alma-wc-refund.php

```diff
@@ -0,0 +1,239 @@
+			/* translators: %s is a username. */
+			$comment_datas['comment_content'] = sprintf( __( 'Order fully refunded via Alma by %s.', 'alma-woocommerce-gateway' ), wp_get_current_user()->display_name );
+		}
+
+		return $comment_datas;
+	}
+
+	/**
+	 * WP action hook "current_screen".
+	 *
+	 * @return void
+	 */
+	public function woocommerce_order_item_add_action_buttons() {
+		if ( is_object( get_current_screen() ) && 'shop_order' === get_current_screen()->id ) {
+			add_filter( 'gettext', array( $this, 'gettext' ), 10, 3 );
+		}
+	}
+
+	/**
+	 * Filters the text of the refund button on a back-office order page.
+	 *
+	 * @param string $translation A text translated.
+	 * @param string $text A text to translate.
+	 * @param string $domain A text domain.
+	 * @return mixed|string
+	 */
+	public function gettext( $translation, $text, $domain ) {
+
+		if ( 'woocommerce' !== $domain || ! array_key_exists( $text, $this->admin_texts_to_change ) ) {
+			return $translation;
+		}
+
+		foreach ( $this->admin_texts_to_change as $original_text => $updated_text ) {
+			if ( $original_text === $text ) {
+				$order = wc_get_order( intval( $_GET['post'] ) ); // phpcs:ignore WordPress.Security.NonceVerification
+				if ( substr( $order->get_payment_method(), 0, 4 ) !== 'alma' ) {
+					return $translation;
+				}
+				$translation = str_replace( $original_text, $updated_text, $text );
+				$this->number_of_texts_changed++;
+			}
+		}
+
+		if ( count( $this->admin_texts_to_change ) === $this->number_of_texts_changed ) {
+			remove_filter( 'gettext', array( $this, 'gettext' ), 10 );
+		}
+
+		return $translation;
+	}
+
+	/**
+	 * Print refund notices.
+	 *
+	 * @return void
+	 */
+	public function admin_notices() {
+
+		if ( 'shop_order' !== get_current_screen()->id ) {
+			return;
+		}
+
+		global $post_id;
+		$refund_notices = get_post_meta( $post_id, 'alma_refund_notices', true );
+
+		if ( ! is_array( $refund_notices ) ) {
+			return;
+		}
+
+		foreach ( $refund_notices as $notice_infos ) {
+			if ( ! is_array( $notice_infos ) ) {
+				continue;
+			}
+			?>
+			<div class="notice notice-<?php echo esc_html( $notice_infos['notice_type'] ); ?> is-dismissible">
+				<p><?php echo esc_html( $notice_infos['message'] ); ?></p>
+			</div>
+			<?php
+		}
+		delete_post_meta( $post_id, 'alma_refund_notices' );
+	}
+
+	/**
+	 * Action hook for order partial refunded.
+	 *
+	 * @param integer $order_id Order id.
+	 * @param integer $refund_id Refund id.
+	 * @return void
+	 */
+	public function woocommerce_order_partially_refunded( $order_id, $refund_id ) {
+
+		$order = wc_get_order( $order_id );
+		if ( true !== $this->refund_helper->is_order_valid_for_partial_refund_with_alma( $order_id, $refund_id ) ) {
+			return;
+		}
+
+		$alma = alma_wc_plugin()->get_alma_client();
+		if ( ! $alma ) {
+			$this->refund_helper->add_order_note( $order_id, 'error', __( 'API client init error.', 'alma-woocommerce-gateway' ) );
+			return;
+		}
+
```

Comment:
you can instanciate WC_Order_Refund directly here and pass it instead of `$refund_id` in the following methods

Reply:
WC_Order_Refund ? Not plutôt WC_Order ?